client/state/jetpack-agency-dashboard/selectors.ts

```diff
@@ -27,8 +27,8 @@ export function checkIfJetpackSiteGotDisconnected( state: AppState ): boolean {
 	return !! state.sites.jetpackSiteDisconnected;
 }
 
-export function getPurchasedLicense( state: AppState ): PurchasedProduct | null {
-	return state.agencyDashboard.purchasedLicense;
+export function getPurchasedLicense( state: AppState ): PurchasedProductsInfo | null {
+	return state.agencyDashboard.purchasedLicense?.purchasedLicenseInfo;
```

Comment:
This is an existing issue where the purchased license isn't displayed in the dashboard when the user is redirected back. It's caused by one of the recent PRs.